authClient/authClient.go

```diff
@@ -66,9 +67,15 @@ func (c *E3dbAuthClient) AuthenticateE3DBClient(ctx context.Context, token strin
 
 // AuthHTTPClient returns an http client that can be used for
 // making authenticated requests to an e3db endpoint using the provided context.
+// The first time AuthHTTPClient is called it use the provided context to create a new http client
+// and will reuse that context until it has expired
 func (c *E3dbAuthClient) AuthHTTPClient(ctx context.Context) *http.Client {
-	if c.httpClient == nil {
+	if c.httpClient == nil || c.ctx.Err() != nil {
+		if c.ctx != nil {
```

Comment:
Isn't the problem the context is expiring thus the http client we create from that context is invalid; wouldn't storing this context and creating a new httpClient from the same expired context lead to the same issue? 

Also I'm a fan of more logging, but I'm not sure what value this gives us here other than an extra log every hour which we can't turn off (or however long it takes to expire); I'd like to see a prefix string added if it stays in `AuthClient context expired: `

Reply:
After re-reading, I amend my edit: We're not refreshing the context here we're just refreshing the client made from a possibly expired context. Wouldn't this lead to the same error?

Reply:
No, this will fix the original error because now instead of the original context expiring and never being refreshed, it will.

Reply:
oh I see, so IF the stored context is expired, use the new context provided and store it; I would argue this is a level of magic caching, bc as a user I'd expect you to use the context I'm passing in but 🤷‍♀

Reply:
Why do we cache the context to begin with, is creating a httpClient from a context expensive?

Reply:
we cache the context to avoid calling auth every time we need to make a service to service call(the code got to its current state because the indexer was bringing auth down)